stdlib/camlinternalMod.ml

```diff
@@ -37,7 +69,7 @@ let rec init_mod loc shape =
       let template =
         Obj.repr (fun _ -> raise (Undefined_recursive_module loc))
       in
-      overwrite closure template;
+      overwrite_closure closure template;
```

Comment:
Right now this is not correct, because  `closure` was obtained by `Obj.new_block`, so all fields are set to `()` and the closure info field is not accurate.  So, `overwrite` is less wrong.  However, `Obj.new_block` should probably be modified to perform minimal initialization of the closure_info field if the tag is Closure_tag.

Reply:
That's a good point and a nice, more general change. I updated `caml_obj_block`:

```c
  if (tg == Closure_tag) {
    /* Closinfo_val is the seconnd field, so we need size at least 2 */
    if (sz <= 2) caml_invalid_argument ("Obj.new_block");
    Closinfo_val(res) = Make_closinfo(0, 2);
  }
```